src/shared/SRP6/SRP6.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the CMaNGOS Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+/// \addtogroup realmd
+/// @{
+/// \file
+
+#ifndef _SRP6_H
+#define _SRP6_H
+
+#include "Common.h"
+#include "Auth/BigNumber.h"
+#include "Auth/Sha1.h"
+#include "ByteBuffer.h"
+
+#include <boost/asio.hpp>
+
+#include <functional>
+
```

Comment:
It looks like a few of these headers are not needed. At least I don't see why we need asio.hpp, ByteBuffer.h and functional here in this header. You're probably missing <string> though (I'm sure it's indirectly included via another header, but perhaps we should not rely on that).